AsusRouterA11yFixes.user.js

```diff
@@ -0,0 +1,266 @@
+			}
+		}
+		if (checkRoot && root.matches(tweak.selector)) {
+			try {
+				applyTweak(root, tweak);
+			} catch (e) {
+				console.log("Exception while applying tweak for '" + tweak.selector + "': " + e);
+			}
+		}
+	}
+}
+
+let observer = new MutationObserver(function (mutations) {
+	for (let mutation of mutations) {
+		try {
+			if (mutation.type === "childList") {
+				for (let node of mutation.addedNodes) {
+					if (node.nodeType != Node.ELEMENT_NODE) {
+						continue;
+					}
+					applyTweaks(node, DYNAMIC_TWEAKS, true);
+				}
+			} else if (mutation.type === "attributes") {
+				applyTweaks(mutation.target, DYNAMIC_TWEAKS, true);
+			}
+		} catch (e) {
+			// Catch exceptions for individual mutations so other mutations are still handled.
+			console.log("Exception while handling mutation: " + e);
+		}
+	}
+});
+
+function init() {
+	applyTweaks(document, LOAD_TWEAKS, false);
+	applyTweaks(document, DYNAMIC_TWEAKS, false);
+	options = { childList: true, subtree: true };
+	if (DYNAMIC_TWEAK_ATTRIBS.length > 0) {
+		options.attributes = true;
+		options.attributeFilter = DYNAMIC_TWEAK_ATTRIBS;
+	}
+	observer.observe(document, options);
+}
+
+/*** Define the actual tweaks. ***/
+
+// Tweaks that only need to be applied on load.
+const LOAD_TWEAKS = [
+	{
+		selector: "#op_link",
+		tweak: el => {
+			const table = el.closest('table');
+			setRole(table, 'banner');
+			// Because I can, make the tbody a list, and each td a list item.
+			setRole(table.firstElementChild, 'list');
+			for (let pres of table.firstElementChild.children) {
+				makePresentational(pres);
+			}
+			// td's become listitems
+			Array.from(table.querySelectorAll('td')).forEach((e) => setRole(e, e.innerText ? 'listitem' : 'none'));
+		},
+	},
+];
+
+// Attributes that should be watched for changes and cause dynamic tweaks to be
+// applied.
+const DYNAMIC_TWEAK_ATTRIBS = [];
+
+// Tweaks that must be applied whenever an element is added/changed.
+const DYNAMIC_TWEAKS = [
+	{
+		selector: '.menu_Desc',
+		tweak: [setRole, 'link'],
+	},
+	{
+		selector: '.menu_Split',
+		tweak: [makeHeading, 2],
+	},
+	{
+		selector: '#mainMenu',
+		tweak: [makeRegion, 'main navigation'],
+	},
+	{
+		selector: '#tabMenu',
+		tweak: [makeRegion, 'secondary navigation'],
+	},
+	{
+		selector: '#tabMenu td',
+		tweak: [setRole, 'link'],
+	},
+	{
+		selector: '.formfonttitle',
+		tweak: [makeHeading, 1],
+	},
+	{
+		selector: 'img[src="/switcherplugin/iphone_switch_container_on.png"]',
+		tweak: e => {
+			e.alt = 'on';
+		},
+	},
+	{
+		selector: "#overDiv_table1",
+		tweak: e => {
+			// on rare occasions, this is delayed while the table renders, so
+			// we wait a quarter second.  Also kind of mimics a tutor help
+			// with most screen readers.
+			setTimeout(() => {
+				announce(e.innerText, 'tutor');
+			}, 250);
+		},
+	},
+];
+
+/** add your specific initialization here, so that if you ever update the framework from new skeleton your inits are not overridden. */
```

Comment:
```suggestion
/** Add your specific initialization here, so that if you ever update the framework from new skeleton your inits are not overridden. */
```